src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResultItem.tsx

```diff
@@ -34,35 +34,51 @@ export class ArtistAuctionResultItem extends Component<Props> {
         {({ user, mediator }) => {
           return (
             <Row>
-              <Responsive>
-                {({ xs, sm, md }) => {
-                  if (xs) {
-                    return (
-                      <ExtraSmallAuctionItem
-                        {...this.props}
-                        mediator={mediator}
-                        user={user}
-                      />
-                    )
-                  } else if (sm || md) {
-                    return (
-                      <SmallAuctionItem
-                        {...this.props}
-                        mediator={mediator}
-                        user={user}
-                      />
-                    )
-                  } else {
-                    return (
-                      <LargeAuctionItem
-                        {...this.props}
-                        mediator={mediator}
-                        user={user}
-                      />
-                    )
-                  }
+              <Responsive2>
+                {breakpoints => {
+                  return (
+                    <>
+                      <breakpoints.xs>
+                        <ExtraSmallAuctionItem
+                          {...this.props}
+                          mediator={mediator}
+                          user={user}
+                        />
+                      </breakpoints.xs>
+
+                      {/*
+                      NOTE:
+
+                      @alloy: I wonder if there's a way to compose breakpoints
+                      togehter. In previous code this was `if (sm || md)`, so
+                      the react JSX below will create exact duplicates of markup
+                      which could be confusing / error prone.
+                    */}
+                      <breakpoints.sm>
+                        <SmallAuctionItem
+                          {...this.props}
+                          mediator={mediator}
+                          user={user}
+                        />
+                      </breakpoints.sm>
+                      <breakpoints.md>
+                        <SmallAuctionItem
+                          {...this.props}
+                          mediator={mediator}
+                          user={user}
+                        />
+                      </breakpoints.md>
+                      <breakpoints.else>
```

Comment:
@damassi is `breakpoints.else` dynamically defined based on adjacent `breakpoints`?

Reply:
Yeah, which gets confusing especially when not nested directly under the Responsive component.